src/dotnet/CarbonAware.LocationSources.Azure/test/AzureLocationSourceTest.cs

```diff
@@ -0,0 +1,97 @@
+using CarbonAware.Model;
+using NUnit.Framework;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Moq.Protected;
+using CarbonAware.Exceptions;
+
+namespace CarbonAware.LocationSources.Azure.Test;
+
+public class AzureLocationSourceTest
+{   
+    [Test]
+    public async Task TestToGeopositionLocation()
+    {
+        var logger = Mock.Of<ILogger<AzureLocationSource>>();
+
+        var mockLocationSource = SetupMockLocationSource().Object;
+        Location inputLocation = new Location {
+            LocationType = LocationType.CloudProvider,
+            CloudProvider = CloudProvider.Azure,
+            RegionName = "eastus"
+        };
+
+        var eastResult = await mockLocationSource.ToGeopositionLocationAsync(inputLocation);
+        AssertLocationsEqual(Constants.LocationEastUs, eastResult);
+
+        inputLocation = new Location {
+            LocationType = LocationType.CloudProvider,
+            CloudProvider = CloudProvider.Azure,
+            RegionName = "westus"
+        };
+
+        var westResult = await mockLocationSource.ToGeopositionLocationAsync(inputLocation);
+        AssertLocationsEqual(Constants.LocationWestUs, westResult);
+
+    }
+
+    // <summary>
+    // If an Azure Location with invalid RegionName is passed, should fail.
+    // </summary>
+    [Test]
+    public void TestToGeopositionInvalidLocation()
+    {
+        var mockLocationSource = SetupMockLocationSource().Object;
+        Location invalidLocation = new()
```

Comment:
Didn't realize you could call `new()` and have it work.  Cool 🎸 

For consistency, however it might be nice to still use `new Location()` like the rest of the file